core/interop/IDynamicInterfaceCastable/src/ManagedApp/NativeObject.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+
+namespace IDynamicInterfaceCastableSample
+{
+    class NativeObject : IDynamicInterfaceCastable, IDisposable
+    {
+        public string Name { get; }
+
+        private delegate int _QueryInterface(IntPtr _this, ref Guid iid, out IntPtr ppv);
+        private delegate uint _Release(IntPtr _this);
+
+        [StructLayout(LayoutKind.Sequential)]
+        private struct IUnknownVtbl
+        {
+            public _QueryInterface QueryInterface;
+            public IntPtr AddRef;
+            public _Release Release;
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        private struct VtblPtr
+        {
+            public IntPtr Vtbl;
+        }
+
+        private class NativeImpl
+        {
+            public IntPtr Ptr;
+            public object Vtbl;
+        }
+
+        private readonly Dictionary<RuntimeTypeHandle, RuntimeTypeHandle> interfaceTypeToImplType;
+        private readonly IntPtr objPtr;
+        private readonly IUnknownVtbl unknownVtbl;
+        private readonly Dictionary<RuntimeTypeHandle, NativeImpl> typeToNativeImpl = new Dictionary<RuntimeTypeHandle, NativeImpl>();
```

Comment:
```suggestion
        private readonly Dictionary<RuntimeTypeHandle, RuntimeTypeHandle> _interfaceTypeToImplType;
        private readonly IntPtr _objPtr;
        private readonly IUnknownVtbl _unknownVtbl;
        private readonly Dictionary<RuntimeTypeHandle, NativeImpl> _typeToNativeImpl = new Dictionary<RuntimeTypeHandle, NativeImpl>();
```

Private fields should be named with a leading `_` and lowercase character.